scripts/gen_offset_header.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017 Intel Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+import argparse
+import sys
+
+def gen_offset_header(input_name, input_file, output_file):
+    obj = ELFFile(input_file)
+
+    symtab = obj.get_section_by_name('.symtab')
+    if not symtab:
+        print("Could not find .symtab section in %s" % input_name)
+        return 1
+
+    if not isinstance(symtab, SymbolTableSection):
+        print(".symtab section is not an a symbol table")
+        return 1
+
+    include_guard = "__INCLUDE_GUARD_%16x__" % abs(input_name.__hash__())
```

Comment:
i think you need %016x to zero-lead the value otherwise there are gonna be spaces